### DISCUSSION TITLE
Some AG Grid events not firing

### DISCUSSION THREAD

**Discussion by thetableman** [zauberzeug/nicegui#3004] [Q&A]
### Question

Consider the below based on the Nice GUI documentation example.

```py
from nicegui import ui

aggrid = ui.aggrid({
    'columnDefs': [
        {'headerName': 'Name', 'field': 'name'},
        {'headerName': 'Age', 'field': 'age'},
    ],
    'rowData': [
        {'name': 'Alice', 'age': 18},
        {'name': 'Bob', 'age': 21},
        {'name': 'Carol', 'age': 42},
    ],
})
aggrid.on('cellClicked', lambda event: ui.notify(f'Cell value: {event.args["value"]}'))
aggrid.on('columnHeaderClicked', lambda event: ui.notify(f'Header value: {event.args["value"]}'))

ui.run()
```

The `cellClicked` event works but the `columnHeaderClicked` doesn't. The Nice GUI [documentation](https://nicegui.io/documentation/aggrid#respond_to_an_ag_grid_event) suggests that all AG Grid `events` should work. Equally the AG Grid [documentation](https://www.ag-grid.com/javascript-data-grid/grid-events/#reference-columns-columnHeaderClicked) lists several 'column' and 'header' based `events` but most, including `columnHeaderClicked`, fail to trigger.

Has anyone had any experience with AG Grid `events` other than `cellClicked`?

**Comment by falkoschindler** [comment]
Hi @thetableman,

That's weird! The "columnHeaderClicked" event doesn't seem to be triggered at all. Sometimes there are problems with serializing the event arguments. But in this case even a pure JavaScript event handler doesn't work:
```py
aggrid.on('columnHeaderClicked', js_handler='console.log(event)')
```

But other events like "columnMoved" do work. So maybe column headers need to be somehow configured "clickable"?

**Comment by thetableman** [comment]
See #3801
